packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,566 @@
+  }
+
+  /**
+   * Get the explorer URL based on the network.
+   *
+   * @returns {string} The explorer URL based on the network.
+   */
+  getExplorerUrl(): string {
+    switch (this.getRadixNetwork()) {
+      case NetworkId.Mainnet:
+        return 'https://dashboard.radixdlt.com'
+      case NetworkId.Stokenet:
+        return 'https://stokenet-dashboard.radixdlt.com'
+      default:
+        throw new Error('Unsupported network')
+    }
+  }
+
+  /**
+   * Get the Gateway URL based on the network.
+   *
+   * @returns {string} The explorer URL based on the network.
+   */
+  getGatewayUrl(): string {
+    switch (this.getRadixNetwork()) {
+      case NetworkId.Mainnet:
+        return MAINNET_GATEWAY_URL
+      case NetworkId.Stokenet:
+        return STOKENET_GATEWAY_URL
+      default:
+        throw new Error('Unsupported network')
+    }
+  }
+
+  /**
+   * Get the explorer URL for a given account address based on the network.
+   * @param {Address} address The address to generate the explorer URL for.
+   * @returns {string} The explorer URL for the given address.
+   */
+  getExplorerAddressUrl(address: Address): string {
+    return `${this.getExplorerUrl()}/account/${address}`
+  }
+
+  /**
+   * Get the explorer URL for a given transaction ID based on the network.
+   * @param {string} txID The transaction ID to generate the explorer URL for.
+   * @returns {string} The explorer URL for the given transaction ID.
+   */
+  getExplorerTxUrl(txID: string): string {
+    return `${this.getExplorerUrl()}/transaction/${txID}`
+  }
+
+  /**
+   *  Validate the given address.
+   * @param {Address} address The address to validate.
+   * @returns {boolean} `true` if the address is valid, `false` otherwise.
+   */
+  async validateAddressAsync(address: string): Promise<boolean> {
+    try {
+      await RadixEngineToolkit.Address.decode(address)
+      return true
+    } catch (error) {
+      return false
+    }
+  }
+
+  validateAddress(): boolean {
+    throw new Error(
+      'validateAddress is synchronous and cannot retrieve addresses directly. Use getAddressAsync instead.',
+    )
+  }
+
+  /**
+   * Helper method to check if a resource is included in a list of assets
+   * @param resource - The resource to be checked
+   * @param assets - The list of assets whret the resource is included
+   * @returns True if the resource is included in the assets list, False otherwise
+   */
+  filterByAssets(resource: any, assets: Asset[]): boolean {
+    return assets.length === 0 || assets.some((asset) => asset.symbol === resource.resource_address)
+  }
+
+  /**
+   * Retrieves the balances of a given address.
+   * @param {Address} address - The address to retrieve the balance for.
+   * @param {Asset[]} assets - Assets to retrieve the balance for (optional).
+   * @returns {Promise<Balance[]>} An array containing the balance of the address.
+   */
+  async getBalance(address: Address, assets: Asset[]): Promise<Balance[]> {
```

Comment:
It doesn't look like this function handles paging.

Reply:
does it need to handle paging?

Reply:
Seems like it might need to if the `assets` are too many, but I'm not sure.

Reply:
I think that other clients probably take care of paging either by having a lower level of their stack handle it (e.g., the SDK they're using) or their APIs are actually not paged.

For us, we would like the user to be able to get the balance of all resources and not the first `page_limit` resources.

Reply:
true. I don't see though how I can use the cursor in the 
```
interface StateEntityDetailsRequest {
    /**
     *
     * @type {LedgerStateSelector}
     * @memberof StateEntityDetailsRequest
     */
    at_ledger_state?: LedgerStateSelector | null;
    /**
     *
     * @type {StateEntityDetailsOptIns}
     * @memberof StateEntityDetailsRequest
     */
    opt_ins?: StateEntityDetailsOptIns;
    /**
     * limited to max 100 items.
     * @type {Array<string>}
     * @memberof StateEntityDetailsRequest
     */
    addresses: Array<string>;
    /**
     *
     * @type {ResourceAggregationLevel}
     * @memberof StateEntityDetailsRequest
     */
    aggregation_level?: ResourceAggregationLevel;
}
```